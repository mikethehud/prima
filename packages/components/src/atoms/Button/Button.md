#### Normal

```jsx
<Button text="Hello world" />{' '}
<Button text="Hello world" color="primary" />{' '}
<Button text="Hello world" color="secondary" />{' '}
<Button text="Hello world" color="success" />{' '}
<Button text="Hello world" color="attention" />
```

#### Outline

```jsx
<Button text="Hello world" type="outline" />
<Button text="Hello world" type="outline" color="primary" />
<Button text="Hello world" type="outline" color="secondary" />
<Button text="Hello world" type="outline" color="success" />
<Button text="Hello world" type="outline" color="attention" />
```

#### Ghost

```jsx
<Button text="Hello world" type="ghost" />
<Button text="Hello world" type="ghost" color="primary" />
<Button text="Hello world" type="ghost" color="secondary" />
<Button text="Hello world" type="ghost" color="success" />
<Button text="Hello world" type="ghost" color="attention" />
```

#### Arrow

A button can have an arrow.

```jsx
<Button text="Hello world" color="primary" arrow="true" />
<Button text="Hello world" color="primary" arrow="true" type="outline" />
<Button text="Hello world" color="primary" arrow="true" type="ghost" />
```
