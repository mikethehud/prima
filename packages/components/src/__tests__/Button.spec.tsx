/* tslint:disable:jsx-no-lambda */

import { shallow } from "enzyme"
import * as React from "react"
import Button from "../Button/Button"
import { ButtonType, ThemeColor } from "../types"

const buttonColors: ThemeColor[] = [
  "default",
  "primary",
  "secondary",
  "success",
  "attention",
  "negative"
]

const buttonTypes: ButtonType[] = ["ghost", "outline", "normal"]

describe("Button", () => {
  it("renders correctly", () => {
    const wrapper = shallow(<Button text="Hello" />)
    expect(wrapper).toMatchSnapshot()
  })

  // Run through all buttonColors
  buttonColors.forEach(buttonColor => {
    it(`can have color ${buttonColor}`, () => {
      const wrapper = shallow(<Button text="Hello" color={buttonColor} />)
      expect(wrapper).toMatchSnapshot()
    })
  })

  // Run through all buttonTypes
  buttonTypes.forEach(buttonType => {
    it(`can be of type ${buttonType}`, () => {
      const wrapper = shallow(<Button text="Hello" type={buttonType} />)
      expect(wrapper).toMatchSnapshot()
    })
  })

  it("can have an arrow", () => {
    const wrapper = shallow(<Button text="Hello" arrow={true} />)
    expect(wrapper).toMatchSnapshot()
  })
})
