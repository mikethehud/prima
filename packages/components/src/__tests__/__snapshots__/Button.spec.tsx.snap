// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button can be of type ghost 1`] = `
<button
  className="button ghost default"
>
  <span>
    Hello
  </span>
</button>
`;

exports[`Button can be of type normal 1`] = `
<button
  className="button normal default"
>
  <span>
    Hello
  </span>
</button>
`;

exports[`Button can be of type outline 1`] = `
<button
  className="button outline default"
>
  <span>
    Hello
  </span>
</button>
`;

exports[`Button can have an arrow 1`] = `
<button
  className="button normal default arrow"
>
  <span>
    Hello
  </span>
  <svg
    viewBox="0 0 512 512"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M294.1 256L167 129c-9.4-9.4-9.4-24.6 0-33.9s24.6-9.3 34 0L345 239c9.1 9.1 9.3 23.7.7 33.1L201.1 417c-4.7 4.7-10.9 7-17 7s-12.3-2.3-17-7c-9.4-9.4-9.4-24.6 0-33.9l127-127.1z"
    />
  </svg>
</button>
`;

exports[`Button can have color attention 1`] = `
<button
  className="button normal attention"
>
  <span>
    Hello
  </span>
</button>
`;

exports[`Button can have color default 1`] = `
<button
  className="button normal default"
>
  <span>
    Hello
  </span>
</button>
`;

exports[`Button can have color negative 1`] = `
<button
  className="button normal negative"
>
  <span>
    Hello
  </span>
</button>
`;

exports[`Button can have color primary 1`] = `
<button
  className="button normal primary"
>
  <span>
    Hello
  </span>
</button>
`;

exports[`Button can have color secondary 1`] = `
<button
  className="button normal secondary"
>
  <span>
    Hello
  </span>
</button>
`;

exports[`Button can have color success 1`] = `
<button
  className="button normal success"
>
  <span>
    Hello
  </span>
</button>
`;

exports[`Button renders correctly 1`] = `
<button
  className="button normal default"
>
  <span>
    Hello
  </span>
</button>
`;
